src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExecuteActionTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using System.Reflection;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using System.Threading;
+using System;
+using System.IO;
+using System.Diagnostics;
+using Microsoft.Diagnostics.Tools.Monitor;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class ExecuteActionTests
+    {
+        private const int TokenTimeoutMs = 2000; // Must be less than DelayMs in the ExecuteApp
+
+        private const string TaskCanceledMessage = "A task was canceled";
+
+        [Fact]
+        public async Task ExecuteAction_ZeroExitCode()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            options.Path = DotNetHost.HostExePath;
+            options.Arguments = GenerateArgumentsString(new string[] { "ZeroExitCode" });
+
+            CollectionRuleActionResult result = await action.ExecuteAsync(options, null, CreateCancellationToken());
+
+            Assert.Equal("0", result.OutputValues["ExitCode"]);
```

Comment:
```suggestion
            Assert.NotNull(result);
            Assert.NotNull(result.OutputValues);
            Assert.True(result.OutputValues.TryGetValue("ExitCode", actualExitCode));
            Assert.Equal("0", actualExitCode);
```

Reply:
Apply this same strategy to all of the other success tests. In fact, make this check a static method so that it can be reused e.g. `private static void ValidateActionResult(CollectionActoinResult result, int expectedExitCode)`

Reply:
Just a note: xUnit complains "Do not use Assert.NotNull() on value type 'CollectionRuleActionResult", so I'll omit that check